subsys/net/ip/net_context.c

```diff
@@ -1558,6 +1569,32 @@ static int get_context_ipv6_v6only(struct net_context *context,
 #endif
 }
 
+static int get_context_recv_pktinfo(struct net_context *context,
+				    void *value, size_t *len)
+{
+#if defined(CONFIG_NET_CONTEXT_RECV_PKTINFO)
+	if (!value || !len) {
+		return -EINVAL;
+	}
+
+	if (*len != sizeof(int)) {
+		return -EINVAL;
+	}
+
+	if (context->options.recv_pktinfo == true) {
+		*((int *)value) = (int) true;
```

Comment:
nit: No need to add space after cast.

Why not cast `context->options.recv_pktinfo` directly, w/o if statement?

Reply:
There are now multiple get option functions in this file, they all follow this same pattern. I think we could create a generic function for these, but in separate PR. I can prepare one after this one is merged.